source/docs/software/advanced-gradlerio/profiling-with-visualvm.rst

```diff
@@ -0,0 +1,168 @@
+Profiling with VisualVM
+=======================
+
+This document is intended to familiarize the reader with the diagnostic tool that is `VisualVM <https://visualvm.github.io/>`__ for debugging Java robot programs. VisualVM is a tool for profiling JVM based applications, such as viewing why an application is using a large amount of memory. This document assumes the reader is familiar with the *risks* associated with modifying their robot `build.gradle`. This tutorial also assumes that the user knows basic terminal/commandline knowledge.
+
+Unpacking VisualVM
+------------------
+
+To begin, `download VisualVM <https://visualvm.github.io/download.html>`__ and unpack it to the WPILib installation folder. The folder is located at ``~/wpilib/`` where ``~`` indicates the users home directory. On Windows, this is ``C:\Users\Public\wpilib``.
```

Comment:
Add options for different OSs via a switcher

Reply:
I don't plan to make OS specific versions of this, Windows is just listed as an explicit example as it's the most used platform.